dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp

```diff
@@ -160,6 +160,10 @@ inline RSOperatorPtr parseTiCompareExpr( //
                     "ColumnRef with field type(" + DB::toString(field_type) + ") is not supported",
                     false);
 
+            // Only support `column` in/not in (literal1, literal2, ...) now.
+            if (expr.children_size() != 2 && child_idx != 0)
```

Comment:
why need to check children_size() != 2 here?

Reply:
for `col in (col)`, `right = OperandType::Unknown`

https://github.com/pingcap/tiflash/blob/971ec1c66f9e3c0e6d94c2a070e0ca0be03f4cf8/dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp#L204-L211